R/translate-sql-base.r

```diff
@@ -175,7 +178,13 @@ base_scalar <- sql_translator(
                         sql(ifelse(side == "both" | side == "right", "RTRIM(", "(")),
                         string
                         ,"))"
-                      )}
+                      )},
+  str_sub = function(string, start = 1L, end = -1L) {
+    start <- as.integer(start)
+    length <- pmax(as.integer(end) - start + 1L, 0L)
+
+    build_sql(sql("SUBSTR"), list(string, start, length))
```

Comment:
I think it's better to use `sql_expr()` for new code

Reply:
I think this also needs a comment briefly explaining the parameterisation of `SUBSTR()`